### ISSUE TITLE
Arithmetic overflow when reading int column

### ISSUE THREAD

**Issue by AaronDavidNewman** [mysql-net/MySqlConnector#1055]
The headline is a little misleading.  It seems the reader is getting very confused wrt which data is going in which columns.  I can see in the result set in the debugger, even the first varchar(50) field is empty.  I didn't see any report of this bug in the issues list.

It is a very perplexing problem because some rows read just fine, others will fail with this specific query but succeed if the query is changed slightly (e.g. add the packageVersion, insert a dummy column in the result set).

I am trying to set up logger now so I can get some better debug information, but here is what I have.

The issue seems to have something to do with the date and int columns in the last 2 fields (lastModified and packageVersion).  I have a similar issue on another table that has a similar shape, but I am using this case because i have tied it to a specific query.  This query is being run as a batch process (not async, not in real time) as part of a prepared statement.

The exception occurs reading the version (second column), but the first field is empty string and that is not what the query is returning when run in workbench with the same parameters.  Below is the text of the query (the text in `MySqlCommand _command`), the table structure, and the stack trace.

This is using AWS RDS (Aurora) MySql 5.7, I just updated the MySqlConnector today to latest stable version (1.3.13),   It is asp.net 4.7 project, if that matters.

``` sql
select x.fkResource,x.version,x.isLatest,x.thumbnailId,x.thumbnailName,x.lastModified 
                from tblResourceVersions x 
                    join tblResource rr on x.fkResource=rr.pkKey and x.version=rr.maxVersion where x.fkResource='<identifier>';
```
tblResourceVersions is like:
``` sql
CREATE TABLE `tblResourceVersions` (
  `fkResource` varchar(50) NOT NULL,
  `version` int(11) NOT NULL,
  `isLatest` bit(1) DEFAULT NULL,
  `thumbnailId` varchar(50) DEFAULT NULL,
  `thumbnailName` varchar(50) DEFAULT NULL,
  `lastModified` datetime DEFAULT NULL,
  `packageVersion` int(11) DEFAULT NULL,
  PRIMARY KEY (`fkResource`,`version`),
  KEY `resourceVersionThumbIx` (`thumbnailId`),
  KEY `tblResourceVersions_ibfk_2` (`packageVersion`),
```
```
"Arithmetic operation resulted in an overflow."
"   at MySqlConnector.Core.TextRow.GetInt32Core(ReadOnlySpan`1 data, ColumnDefinitionPayload columnDefinition) in /_/src/MySqlConnector/Core/TextRow.cs:line 33
   at MySqlConnector.Core.Row.GetInt32(Int32 ordinal) in /_/src/MySqlConnector/Core/Row.cs:line 231
   at MySqlConnector.MySqlDataReader.GetInt32(Int32 ordinal) in /_/src/MySqlConnector/MySqlDataReader.cs:line 242
   at MySqlConnector.MySqlDataReader.GetFieldValue[T](Int32 ordinal) in /_/src/MySqlConnector/MySqlDataReader.cs:line 376
   at SqlRepository.ImportRepository.MoveItem(String itemGuid, ImportParameters environment) in C:\\src\\bb\\airways\\authoring_testauthoring\\SqlRepository\\ImportRepository.cs:line 154

```

![image](https://user-images.githubusercontent.com/5438280/136454711-8380882f-43ff-444f-b325-9673efc71625.png)

```

**Comment by bgrainger** [comment]
If you're using prepared commands (`MySqlCommand.Prepare`), then this is very likely a duplicate of #1018.

That's fixed in 1.4.0-beta.4 (which should be stable enough to use), but hasn't been backported to a 1.3.x release (yet).

**Comment by bgrainger** [comment]
> I downgraded to 1.0 and that is where it worked.

That's because in 1.2 (and earlier), `Ignore Prepare = True` is on by default in the connection string, so `MySqlCommand.Prepare()` does nothing. (So the bug is still there, but it's not exposed.)
